### PR TITLE
[RCM-374] Add RC db to the flare

### PR DESCRIPTION
**Pull request by coignetp**
### What does this PR do?

Add remote-config.db to the flare, but without the target files.

### Describe how to test/QA your changes

* Start an agent with RC enabled
* Start a tracer requesting any RC product
* Run `agent flare`
* Make sure there is a `remote-config.db` file in the zip flare
* Make sure this file does not contain target files

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/flare/remote_config.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+	}
+
+	dstDB, err := bbolt.Open(dstPath, 0600, &bbolt.Options{})
+	if err != nil {
+		os.Remove(dstPath)
+		return err
+	}
+	defer dstDB.Close()
+
+	err = dstDB.Update(func(tx *bbolt.Tx) error {
```
Comment (mellon85) [inline pkg/flare/remote_config.go:46]: As it's a read only operation we can use dstDB.View

File: pkg/flare/remote_config.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	dstPath := filepath.Join(tempDir, hostname, "remote-config.db")
+	srcPath := filepath.Join(config.Datadog.GetString("run_path"), "remote-config.db")
+
+	err := util.CopyFileAll(srcPath, dstPath)
+	if err != nil {
+		// Prevent from having a clear db here
+		os.Remove(dstPath)
+		return err
+	}
+
```
Comment (mellon85) [inline pkg/flare/remote_config.go:38]: can we add here a `defer os.Remove`? it will simplify the rest of the code

Comment (coignetp) [reply pkg/flare/remote_config.go:38]: We don't always want to delete the file at the end of the function, only when something goes wrong and we couldn't process the flare db

**Review by mellon85 — APPROVED** [review]

**Review by djmitche — APPROVED** [review]
Looks good, assuming that the DB has no secret information that customers wouldn't want in a flare.

File: pkg/flare/remote_config.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util"
+	"go.etcd.io/bbolt"
+)
+
+func stripRCTargets(raw []byte) []byte {
```
Comment (cedricvanrompay-datadog) [inline pkg/flare/remote_config.go:20]: nitpicking: the name of this function should be "hash" not "strip"

**Review by cedricvanrompay-datadog — APPROVED** [review]
Looks good to me assuming all targets are indeed in buckets which name ends with `_targets`.

Did you test with an actual `remote-config.db`?

**Pull request merged**